src/pages/proposal.tsx

```diff
@@ -57,59 +65,112 @@ import './proposals.css'
 
 // TODO: Review why proposals.css is being imported and use only proposal.css
 
+export type SelectedChoice = { choice?: string; choiceIndex?: number }
+const EMPTY_CHOICE_SELECTION: SelectedChoice = { choice: undefined, choiceIndex: undefined }
 const PROPOSAL_STATUS_WITH_UPDATES = new Set([ProposalStatus.Passed, ProposalStatus.Enacted])
+const EMPTY_CHOICES: string[] = []
+const MAX_ERRORS_BEFORE_SNAPSHOT_REDIRECT = 3
+const SECONDS_FOR_VOTING_RETRY = 5
 
-type ProposalPageOptions = {
-  changing: boolean
+export type ProposalPageContext = {
```

Comment:
I'm not sure about naming this Context as Context in a react app _context_ is a core react API feature... 🤔

Reply:
I think I prefer state as that's what this is. (`ProposalPageState`)